bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableOptionsFactory.java

```diff
@@ -267,10 +305,29 @@ public static BigtableOptions fromConfiguration(Configuration configuration) thr
           reportDirectoryPath.resolve("call_timing.txt").toAbsolutePath().toString();
       LOG.debug("Logging call status aggregates to %s", callStatusReport);
       LOG.debug("Logging call timing aggregates to %s", callTimingReport);
-      optionsBuilder.setCallStatusReportPath(callStatusReport);
-      optionsBuilder.setCallTimingReportPath(callTimingReport);
+      channelOptionsBuilder.setCallStatusReportPath(callStatusReport);
+      channelOptionsBuilder.setCallTimingReportPath(callTimingReport);
     }
 
+    channelOptionsBuilder.setRetryOptions(createRetryOptions(configuration).build());
+
+    int channelCount =
+        configuration.getInt(BIGTABLE_CHANNEL_COUNT_KEY, BIGTABLE_CHANNEL_COUNT_DEFAULT);
+    channelOptionsBuilder.setChannelCount(channelCount);
+
+    long channelTimeout =
+        configuration.getLong(BIGTABLE_CHANNEL_TIMEOUT_MS_KEY, BIGTABLE_CHANNEL_TIMEOUT_MS_DEFAULT);
+
+    Preconditions.checkArgument(channelTimeout == 0 || channelTimeout >= 60000,
```

Comment:
Where does this restriction come from? Add comment?
The message should also mention that it can be set to 0.